### ISSUE TITLE
[Bug]: Inpaint and no-half

### ISSUE THREAD

**Issue by SunGreen777** [lshqqytiger/stable-diffusion-webui-directml#300]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits

### What happened?

Help me, please.
I own a radeon rx570 8gb.
-no-half, does not allow rendering at high resolutions
With -no-half missing, inpaint doesn't work.
The alternative --upcast-sampling has no effect.
Win 11, 32ram

### Steps to reproduce the problem

Remove the -no-half argument, check that inpaint works
Add argument -no-half try to create image 768x768 or +

### What should have happened?

Inpaint should work and render at high resolutions

### Sysinfo

A1111
Windows 11 - 32ram
[sysinfo-2023-10-10-14-01.txt](https://github.com/lshqqytiger/stable-diffusion-webui-directml/files/12858092/sysinfo-2023-10-10-14-01.txt)

### What browsers do you use to access the UI ?

Other

### Console logs

```Shell
--- go Additional information >>
```

### Additional information

There are no errors  console,
Without a no-half argument.  In UI no result,  or  have spotting
![image](https://github.com/lshqqytiger/stable-diffusion-webui-directml/assets/61922120/969b59bf-f6a3-473a-93cc-adac407bbc4c)

**Comment by Ptibouc77** [comment]
It's a known issue you need `--no-half` to inpaint (also you need to set negative guidance sigma to 0 if it set to higher it also doesn't work)
I think that 768x768 is a bit too much for your GPU
For your rx570 try with`--medvram --precision full --no-half --no-half-vae`


I'm on a rx6600 i can do txt2img at 560x840 (i can go a little bit higher but it will take too long for me) with `--medvram --opt-sdp-attention --upcast-sampling` and i need to switch back to `-no-half  `if i want to inpaint.

**Comment by SunGreen777** [comment]
I put two .bats with different configs in SD. (they can be renamed)
One for Inpaint the second for generation
1280/720px  OK

1)
*
@echo off
set PYTHON=
set GIT=
set VENV_DIR=
set GIT_SSL_NO_VERIFY=true
git pull
set COMMANDLINE_ARGS= --medvram --opt-sdp-no-mem-attention --api
set SAFETENSORS_FAST_GPU=1
call webui.bat

2) - inpaint
*
@echo off
set PYTHON=
set GIT=
set VENV_DIR=
set GIT_SSL_NO_VERIFY=true
git pull
set COMMANDLINE_ARGS= --opt-sub-quad-attention --medvram --api --no-half --disable-nan-check
set SAFETENSORS_FAST_GPU=1
call webui.bat

**Comment by vufper45** [comment]
I have same issue. Unfortunatelly, i don't remember the version i was on but i installed SD on May 20 and never did git pull after. I was able to render and inpaint up to 512x768 with argument --medvram --precision full --no-half. Somewhere in September i decided to do git pull and with all improvements i noticed that with same arguments the vram usage increased drastically. I tried different set of arguments (--opt-split-attention --no-half-vae --disable-nan-check) but it broke inpainting and ADetailer, i hope this will be fixed someday because it is tricky to change arguments every time i want to inpaint and set resolution about 400x400 max (i have rx6600m 8 gb)

**Comment by songib** [comment]
Yep, It's been there since the beginning of time. It's directML issue I think.

**Comment by xhy2008** [comment]
I have the same problem,both in img2img and in/outpainting.
My GPU has only 4GB of VRAM,so no-half is not able to use.

**Comment by mr-september** [comment]
Just adding another possibly useless report that upcast sampling kills inpainting with AMD 5700, directml and Windows.

Would be nice if there was more visibility around this incompatibility, for example mentioned in the [arguments wiki](https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Command-Line-Arguments-and-Settings)
